kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala

```diff
@@ -147,6 +163,36 @@ object Utils extends Logging {
     dir
   }
 
+  /**
+   * write InputStream to temp file in directory
+   * @param is inputstream for file
+   * @param dir dir path for temp file creation
+   * @param fileName original file name with suffix
+   * @return created new temp file in dir in file name with random path
+   */
+  def writeToTempFile(is: InputStream, dir: Path, fileName: String): File = {
+    try {
+      if (!dir.toFile.exists()) {
+        dir.toFile.mkdirs()
+      }
+      val (prefix, suffix) = fileName.lastIndexOf(".") match {
+        case i if i > 0 => (fileName.substring(0, i), fileName.substring(i))
+        case _ => (fileName, "")
+      }
+      val filePath = Paths.get(dir.toString, s"$prefix-${abs(Random.nextInt())}$suffix")
```

Comment:
I recommend to use the below pattern.

```
rootDir/sessionId/fileName
```

Reply:
otherwise, it is difficult to scale out if we want to support upload more files not only resource file.

Reply:
BTW, now the pattern with random, I wonder there might be duplicate issue.

Reply:
and delete the `rootDir/sessionId` dir finally instead of delete the single file.

Reply:
seems you can get the sessionId here.

Then I recommend to use UUID instead of random int.

Reply:
cc @pan3793

Reply:
UUID might be too long for the random identifier here, considering the original filename is preserved in new file name.
To avoid dumplication issue and increase readiblity, 
how about `rootDir/${original_preffix}-${datetime_YYYYMMDDHHMMSS}-${randomInt}.${original_suffix}`?
@pan3793 @turboFei

Reply:
I recommend you reserve an accumulator instead of randomInt.

Likes yarn applicationId(application_$clusterTimestamp_$accId).

Reply:
randomInt is meaningless.

Reply:
The accumulator is a better option, while it also could be cleaned up after server's shutdown.

Reply:
It does not matter， there is a timestamp.

Reply:
Addressed. Turned to using an atomic counter for file name identifier generation instead.